src/onnx/parse_resize.cpp

```diff
@@ -192,71 +193,117 @@ static std::vector<double> get_scales(const onnx_parser::attribute_map& attr)
-            check_arg_empty(arg_out_s,
-                            "PARSE_" + op_name + ": dynamic output size is not supported!");
+            if(arg_out_s.empty())
+                return true;
             arg_out_s.visit([&](const auto& ol) { out_lens.assign(ol.begin(), ol.end()); });
 
             if(out_lens.size() != in_lens.size())
             {
                 MIGRAPHX_THROW("PARSE_" + op_name +
-                               ": specified output size does not match input size");
+                               ": specified output size's rank does not match input size");
             }
 
-            // compute the scale
+            // compute the scales
             vec_scale.resize(in_lens.size());
             std::transform(in_lens.begin(),
                            in_lens.end(),
                            out_lens.begin(),
                            vec_scale.begin(),
                            [](auto iss, auto oss) { return 1.0 * oss / iss; });
+            return false;
         }
         else
         {
-
-            // scale input
+            // this argument is scale input
             if(lens[0] == in_lens.size())
             {
                 auto arg_scale = arg->eval();
-                check_arg_empty(arg_scale,
-                                "PARSE_" + op_name + ": dynamic input scale is not supported!");
+                if(arg_scale.empty())
+                    return true;
 
                 arg_scale.visit([&](const auto& v) { vec_scale.assign(v.begin(), v.end()); });
             }
+            return false;
         }
     }
+    MIGRAPHX_THROW("PARSE_" + op_name + ": no shapes or scales input provided");
 }
 
 struct parse_resize : op_parser<parse_resize>
 {
     std::vector<op_desc> operators() const { return {{"Resize"}, {"Upsample"}}; }
 
+    // Helper to add a "reshape" and "gather" instruction.  These can implement
+    // Nearest mode resizing if all sizes are known at compile time.
+    instruction_ref make_gather_instruction(const onnx_parser::node_info& info,
+                                            const std::size_t out_elements,
+                                            const shape& in_s,
+                                            shape& out_s,
+                                            const std::vector<size_t>& in_lens,
+                                            const std::vector<size_t>& out_lens,
+                                            const std::vector<double>& vec_scale,
+                                            instruction_ref args_0) const
+    {
+        std::string nearest_mode = get_nearest_mode(info.attributes);
+        std::vector<int> ind(out_elements);
+
+        // map out_idx to in_idx
+        auto nearest_op              = get_nearest_op(nearest_mode);
+        std::string coord_trans_mode = get_coord_trans_mode(info.attributes);
+        auto idx_op                  = get_original_idx_op(coord_trans_mode);
```

Comment:
Can we use the functions in the `resize` operator so we don't have repeated code?

Reply:
done.